### DISCUSSION TITLE
Shadow: how to change size?

### DISCUSSION THREAD

**Discussion by dkrnl** [google/model-viewer#2814] [Q&A] [answered]
Hi!

Problem: some shadows are cropped (see attached picture).
Question: How can i change shadow size (floor plane size) ?

![image](https://user-images.githubusercontent.com/734622/134274639-6e99c710-4bf8-46b2-aebb-4b86388306c9.png)

**Comment by elalish** [comment]
That seems like a bug; we set the shadow plane size internally to avoid this kind of thing. Is this model animated? That's one difficult case, as it's hard to calculate the size of the scene. Can you give a repro link?

**Comment by dkrnl** [comment, marked as answer]
Hi!
Sorry for delay in reply. Model in attachment archive: [figura_new_1.zip](https://github.com/google/model-viewer/files/7222849/figura_new_1.zip)

**Reply by elalish** [reply]
Ah, indeed the problem is the animation; we  make the shadow plane 3x larger than the  resting pose because we can't tell how large the model might get on any given frame (and recalculating it each frame is  very expensive). I would recommend adding some tiny invisible fixed geometry at the  bounds of your largest animation frame (degenerate triangles for instance). That  will allow us to know the full size of the scene initially and size things properly.

**Reply by dkrnl** [reply]
@elalish thank you very much!
